src/OrleansEventSourcing/EventStorage/LogViewAdaptor.cs

```diff
@@ -0,0 +1,373 @@
+
+            if (this.provider == null)
+            {
+                throw new ArgumentException("provider must be an event store", nameof(provider));
+            }
+        }
+
+        // streamname must be constructed lazily because grain reference is not available at construction time
+        private string streamName
+        {
+            get
+            {
+                return EventStream.StreamName;
+            }
+        }
+
+        protected override TState LastConfirmedView()
+        {
+            return confirmedState;
+        }
+
+        protected override int GetConfirmedVersion()
+        {
+            return confirmedVersion;
+        }
+
+        protected override void InitializeConfirmedView(TState initialState)
+        {
+            confirmedState = initialState;
+            confirmedVersion = 0;
+        }
+
+        public override async Task<IReadOnlyList<TEvent>> RetrieveLogSegment(int fromVersion, int toVersion)
+        {
+            var response = await EventStream.Load<TEvent>(fromVersion, toVersion);
+
+            if (response.ToVersion != toVersion || response.FromVersion != fromVersion || response.Events.Count != (toVersion - fromVersion))
+            {
+                throw new NotSupportedException("the event store returned a different segment than was requested");
+            }
+
+            return response.Events.Select(kvp => (TEvent) kvp.Value).ToList();
+        }
+
+        /// <summary> Creates a submission entry used to track this event </summary>
+        protected override SubmissionEntryWithGuid<TEvent> MakeSubmissionEntry(TEvent entry)
+        {
+            return new SubmissionEntryWithGuid<TEvent>()
+            {
+                Entry = entry,
+
+                // for any events that do not already have a Guid, create one
+                Guid = (entry as IEventWithGuid)?.Id ?? Guid.NewGuid()
+            };
+        }
+
+        // Read the latest primary state, retrying until successful
+        protected override async Task ReadAsync()
+        {
+            await ReadAsyncInternal();
+        }
+
+        // Read the latest primary state, retrying until successful
+        // if a guid is specified, while doing so, look for it in the stream
+        private async Task<bool> ReadAsyncInternal(Guid? lookFor = null)
+        {
+            bool guid_was_seen = false;
+
+            EventStreamSegment<TEvent> response;
+
+            while (true)
+            {
+                try
+                {
+                    Services.Log(Severity.Verbose, "Read issued for position={0}", confirmedVersion);
+
+                    response = await EventStream.Load<TEvent>(confirmedVersion);
+
+                    break;
+                }
+                catch (Exception e)
+                {
+                    LastPrimaryIssue.Record(new LoadEventsFailed() { Exception = e }, Host, Services);
+
+                    await LastPrimaryIssue.DelayBeforeRetry();
```

Comment:
Using something like Polly would allow us more flexibility with fine-tuning the strategy, as well as bundling a circuit breaker with it, so that even if they individually retry forever, we give both systems some time to breadth if communications between the 2 are going very badly